pkg/providers/aws/provider_redis.go

```diff
@@ -370,6 +390,15 @@ func (p *AWSRedisProvider) deleteElasticacheCluster(cacheSvc elasticacheiface.El
 		return croType.StatusMessage(fmt.Sprintf("delete detected, deleteReplicationGroup() in progress, current aws elasticache status is %s", *foundCache.Status)), nil
 	}
 
+	namespace := r.Namespace
```

Comment:
Same as above, I would lean toward brevity here and not declare these variables and pass them directly to the functions